titan-compiler/coder.lua

```diff
@@ -1008,18 +1104,15 @@ generate_stat = function(stat, ctx)
                 stat.exp._type, var_lvalue.slot_address, exp_cvalue,
                 var_lvalue.parent_pointer)
             assign_stat = util.render([[
-                if (isempty(${SLOT})) {
-                    luaL_error(L,
-                        "out of bounds (inside array part) at line %d",
-                        ${LINE});
-                }
+                if (!${CHECK_TAG}) { titan_runtime_array_type_error( L, ${LINE}, ${EXPECTED_TAG}, ${SLOT} ); }
```

Comment:
Could we break the line here to maintain a pattern?
Also, no need for the extra spaces between the parenthesis.

Reply:
I put it on a single line because I found it much easier to work with the generated C code when it was that way. The group of statements behind an array access became more visually distinct and it was much easier to comment the check in and out...

Reply:
Could we change the other error checks to be one liners too?